pkg/resource/resource.go

```diff
@@ -0,0 +1,118 @@
+package resource
+
+import (
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/catalog"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/controller"
+	"github.com/IBM-Cloud/bluemix-go/api/resource/resourcev1/management"
+	"github.com/IBM-Cloud/bluemix-go/models"
+	"github.com/IBM-Cloud/bluemix-go/session"
+	"k8s.io/klog/v2"
+	"os"
+)
+
+func CreateServiceInstance(c *session.Session, Name, ServiceName, ServicePlan, ResourceGrp, Region, apikey string) (string, error) {
+	catalogClient, err := catalog.New(c)
+
+	if err != nil {
+		return "", err
+	}
+	resCatalogAPI := catalogClient.ResourceCatalog()
+
+	service, err := resCatalogAPI.FindByName(ServiceName, true)
+	if err != nil {
+		return "", err
+	}
+
+	servicePlanID, err := resCatalogAPI.GetServicePlanID(service[0], ServicePlan)
```

Comment:
referring the index value directly here is always risky, we need to think about how to avoid this,